addons/mail/static/src/models/thread/thread.js

```diff
@@ -854,6 +876,15 @@ function factory(dependencies) {
             this.update({ custom_channel_name: newName });
         }
 
+        /**
+         * Change the pinned state of this thread and notify server of the change.
+         * @param {boolean} isPinned the new pinned state
+         */
+        async setIsPinned(isPinned) {
```

Comment:
We usually have more functional method names (eg. "pin" and "unpin" instead of "setters" like this). Not sure what we want exactly here? @alexkuhn

Reply:
`changePinState` ?

Reply:
`updateIsPinned`?

Reply:
You don't like having methods being simply called "pin" and "unpin"?

A user would explain his intent as "(un)pin this thread" not "change the pin state of this thread to true/false" which are technical. User shouldn't even be aware pinned is stored as boolean.

Reply:
Having `pin()` and `unpin()` methods is good